docs/core-concepts.md

```diff
@@ -32,14 +30,12 @@ Here is a sample deployment config for the EC2 operator.
 
 1 :- `api_version` specifies deployment configuration version.
 
-2 :- `metadata` contains the base information about the deployment.
+2 :- `name` Deployment's name. bentoctl recommends to keep deployment name unqiue within each operator to avoid any potential issues.
 
-3 :- `spec` specifics the deployment details. The deployment detail options are provided by the operator listed in the metadata section.
+3 :- `operator` Operator used for the deployment. bentoctl will automatically install official operator if it is not installed.
 
-4 :- `metadata.name` - Deployment's name. bentoctl recommends to keep deployment name unqiue within each operator to avoid any potential issues.
+4 :- `template` The template for the deployment. It determines what the generated terraform project will look like.
```

Comment:
is it `template` or `template_type`? The code currently uses `template_type` but just template does look clean.

Reply:
it is `template`. It is updated in the deployment config class